### PR TITLE
Significant re-working of the setup script

### PR DESCRIPTION
**Pull request by justinclift**
Took some time today to re-work the setup.sh script.  There's a bunch of improvements in it that are really overdue:

* Friendly text output
* Support for more than just Ubuntu
  * Debian support is in this PR.  Tested on Debian 12.x (bookworm) release
  * Will likely add others (RHEL compatible, SLES) in the near future
* Doesn't automatically add redash to the Docker Compose environment variables in the users profile
  * It's a source of pain for some users (https://github.com/getredash/setup/issues/48)

---

To try out this script, download it then run it as root:

```
# wget https://github.com/getredash/setup/raw/debian_non_sudo_v1/setup.sh
# sh setup.sh
```

You can probably run it via sudo as well, not that I've tried that yet:

```
$ wget https://github.com/getredash/setup/raw/debian_non_sudo_v1/setup.sh
$ sudo ./setup.sh
```

---

Note that this PR has an extra commit added on the end to override the branch name.  It's just needed to make this script work while it's not in the `master` branch.

We'll need to manually remove that commit before we merge this.

### CODE REVIEW

**Comment by justinclift** [comment]
@getredash/maintainers FYI.  I'll probably merge this tomorrow if I don't hear back from anyone. :smile:

File: setup.sh
Comment (lucasfcnunes) [inline setup.sh]: We could use `compose.yaml` here (and all other places)

https://github.com/getredash/redash/commit/af2f4af8a24165edc65d433ec25ef1462a563d17 (it's in already in `redash/master`)

File: setup.sh
Comment (lucasfcnunes) [inline setup.sh:120]: Change service `data/docker-compose-up.service` to use `docker compose` too?

https://github.com/getredash/setup/blob/69e01877fe1a065d53e1018cb607256a54ed9ff7/data/docker-compose-up.service#L9

File: setup.sh
Comment (lucasfcnunes) [inline setup.sh:61]: After installing should we set the `docker` group so users can use `docker` without `sudo`?


Like:
```sh
groupadd docker || true # don't fail to add if it already exists
usermod -aG docker $USER
newgrp docker
```

https://docs.docker.com/engine/install/linux-postinstall

**Review by lucasfcnunes — CHANGES_REQUESTED** [review]

File: setup.sh
Comment (justinclift) [reply setup.sh]: Good idea.  That's not yet changed in master, but should be once [this PR](https://github.com/getredash/redash/pull/6674) is merged.

File: setup.sh
Comment (justinclift) [reply setup.sh:120]: Oops, didn't even think to look at that file.  Will take a proper look at it tomorrow and change that + anything else that seems relevant. :smile:

File: setup.sh
Comment (justinclift) [reply setup.sh:61]: Normally I'd say yes, but in this one specific case the script is designed to be run as the root user.  Either directly, or through sudo.  So it would effectively be a no-op if we added it back in. :wink:

The previous script, which embedded sudo's throughout it, was able to better add the user to the `docker` group as you suggest.

Am open to ideas if there's some way of making it work... without adding a bunch of sudo's back into the script. :smile:

File: setup.sh
Comment (justinclift) [reply setup.sh]: Oh, is it `compose.yaml` or `compose.yml`?  That PR in the Redash repo uses the `yml` extension.  Seems like something we'd probably want to match up with.

Comment (lucasfcnunes) [reply setup.sh]: You guys are deleting the `compose.yaml`.

image

Comment (lucasfcnunes) [reply setup.sh]: *`.yaml` is prefered over `.yml`

File: redash_make_default.sh
Comment (justinclift) [inline redash_make_default.sh]: If we want to carry over the `$REDASH_BASE_PATH` value from the setup script, we could probably change that `/opt/redash` text fragment into a token and have the setup script replace it.

I'll probably add that in tomorrow too, :smile:

File: setup.sh
Comment (justinclift) [reply setup.sh:61]: I guess we could add some kind of post install instruction about it, so the user is able to add themself to the group if appropriate.

File: setup.sh
Comment (lucasfcnunes) [reply setup.sh]: > Oh, is it `compose.yaml` or `compose.yml`? That PR in the Redash repo uses the `yml` extension. Seems like something we'd probably want to match up with.

Here are some references I personally use:
- https://docs.docker.com/compose/compose-application-model/#the-compose-file:~:text=The%20default%20path,(preferred) `compose.yaml`
- https://yaml.org/faq.html#:~:text=Please%20use%20%22.yaml%22%20when%20possible. `.yaml`

Comment (justinclift) [reply setup.sh]: Cool, thanks. :smile:

File: redash_make_default.sh
Comment (justinclift) [reply redash_make_default.sh]: Done with the latest commit in this PR.

File: setup.sh
Comment (justinclift) [reply setup.sh:120]: Done. :smile:

File: setup.sh
Comment (justinclift) [reply setup.sh]: Done that rename (in this PR). :smile:

File: setup.sh
Comment (justinclift) [reply setup.sh:61]: Been thinking this over, and for now lets also skip adding a post install text piece about this.  The post-install will start to look like a wall of text otherwise, which will mean people won't read it.

If it turns out that was a dumb idea on my part and we should add it after all, then we can do so then. :smile:

**Comment by justinclift** [comment]
Should be ok to merge (MANUALLY, after removing the branch override name commit) now.

Just finished testing it on Debian 12.x, Ubuntu 20.04, and Ubuntu 22.04.

"Testing" in this case involved running the setup.sh script and verifying Redash installed ok and the webUI is functional (aka initial login, nothing seems broken).

**Comment by justinclift** [comment]
Just manually merged all of the commits into one, except for the branch name override.  That should make things super easy to merge (manually, from the cli). :smile:

**Comment by justinclift** [comment]
k, I'm going to merge this now. :smile:

**Comment by justinclift** [comment]
And managed to press the wrong button while I was at it.  Anyway, it's merged now.  So people using the setup script can also do it on Debian 12.5 rather than Ubuntu.

**Comment by lucasfcnunes** [comment]
Thanks for your work, @justinclift!

**Comment by justinclift** [comment]
@lucasfcnunes All good. :smile:

There's a [follow up PR](https://github.com/getredash/setup/pull/59) which adds support for RHEL and its clones if you're up for taking a look over it?

That one's essentially complete, though the commit history in that is getting a bit messy now.  I'll clean it up properly before merging of course. :smile:

**Pull request merged**
